pandas/plotting/_matplotlib/converter.py

```diff
@@ -826,6 +828,7 @@ def _monthly_finder(vmin, vmax, freq: BaseOffset) -> np.ndarray:
     return info
 
 
+@functools.cache
```

Comment:
No chance of getting non-hashable inputs?

Reply:
I think vmax/vmin are always floats. I'll type these so we have some validation